src/sideeffect.d

```diff
@@ -313,10 +313,10 @@ extern (C++) void discardValue(Expression e)
              */
             if (!lambdaHasSideEffect(ce.e1) && !lambdaHasSideEffect(ce.e2))
             {
-                discardValue(ce.e1);
-                discardValue(ce.e2);
+                return discardValue(ce.e1) |
```

Comment:
@yebblies well `|` is more in keeping with the old behavior (print both messages)